app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/AbstractSampleProvider.java

```diff
@@ -90,6 +99,29 @@ public void addGBActivitySamples(T[] activitySamples) {
         getSampleDao().insertOrReplaceInTx(activitySamples);
     }
 
+    @Override
+    public void exportToCSV(T[] activitySamples, File outFile) {
+        String separator = ",";
+
+        LOG.info("Exporting samples into csv file: " + outFile.getName());
+        try {
+            BufferedWriter bw = new BufferedWriter(new FileWriter(outFile));
+
+            for (T sample : activitySamples){
+            String line = sample.getTimestamp() + separator + sample.getDeviceId() + separator + sample.getUserId() + separator + sample.getRawIntensity() + separator + sample.getSteps() + separator + sample.getRawKind() + separator + sample.getHeartRate();
+
+            //LOG.debug("Adding line into buffer: " + line);
+            bw.write(line);
+            bw.newLine();
+        }
+
+            bw.flush();
+            bw.close();
+            } catch (IOException e) {
+            LOG.error(e.getMessage());
```

Comment:
Please use
`LOG.error("Explain what failed", e);`
otherwise, the stacktrace of the exception will not be logged, making debugging hard.